geode-core/src/main/java/org/apache/geode/cache/query/internal/QueryConfigurationServiceImpl.java

```diff
@@ -141,12 +156,18 @@ public void updateMethodAuthorizer(Cache cache, String className, Set<String> pa
               String.format(INTERFACE_NOT_IMPLEMENTED_MESSAGE, userClass.getName(),
                   MethodInvocationAuthorizer.class.getName()));
         }
+
         MethodInvocationAuthorizer tmpAuthorizer =
             (MethodInvocationAuthorizer) userClass.newInstance();
         tmpAuthorizer.initialize(cache, parameters);
         this.authorizer = tmpAuthorizer;
-
       }
+
+      // Invalidate cached CQ results, if any.
+      cqService.getAllCqs().forEach(cqQuery -> {
```

Comment:
Should this be pulled into a method?  It would be self documenting (no comment needed) if the method name was correctly chosen

Reply:
Done!